src/Http/Http.Results/src/Results.cs

```diff
@@ -495,61 +495,61 @@ public static IResult RedirectToRoute(string? routeName = null, object? routeVal
     /// <param name="statusCode">The status code to set on the response.</param>
     /// <returns>The created <see cref="StatusCodeHttpResult"/> object for the response.</returns>
     public static IResult StatusCode(int statusCode)
-        => new StatusCodeHttpResult(statusCode);
+        => ResultsCache.StatusCode(statusCode);
 
     /// <summary>
     /// Produces a <see cref="StatusCodes.Status404NotFound"/> response.
     /// </summary>
     /// <param name="value">The value to be included in the HTTP response body.</param>
     /// <returns>The created <see cref="IResult"/> for the response.</returns>
     public static IResult NotFound(object? value = null)
-        => new NotFoundObjectHttpResult(value);
+        => value is null ? ResultsCache.NotFound : new NotFoundObjectHttpResult(value);
 
     /// <summary>
     /// Produces a <see cref="StatusCodes.Status401Unauthorized"/> response.
```

Comment:
Sorry, I am confusing about your question. This will return a cached `NotFoundObjectHttpResult` not `UnauthorizedHttpResult`. Maybe I am missing something.

Reply:
My comment is about the `<summary/>` after the overload that returns `NotFoundObjectHttpResult`. I probably should have commented on that earlier overload.

In any case, I was confused and thought the `<summary/>` was about the return type. `StatusCodes.Status404NotFound`, `StatusCodes.Status401Unauthorized`, et cetera are of course `int`s not `IResult`s.

In retrospect, my suggestion should have been to make the return type (`NotFoundObjectHttpResult`, `UnauthorizedHttpResult`, et cetera) clear _as well_.

Reply:
I agree about the return type but that is a problem that we have right now because this class was shipper already but we are working on it https://github.com/dotnet/aspnetcore/issues/41009

Reply:
My suggestion was about the doc comments, not the method signature e.g.
```suggestion
    /// Returns an <see cref="UnauthorizedHttpResult"/> that produces a <see cref="StatusCodes.Status401Unauthorized"/> response.
```

Again, this is a suggestion and could be done later. You're not otherwise changing return types or updating `<summary/>` elements in this PR.